source/common/http/utility.cc

```diff
@@ -1458,6 +1458,10 @@ Utility::convertCoreToRouteRetryPolicy(const envoy::config::core::v3::RetryPolic
 
 bool Utility::isSafeRequest(const Http::RequestHeaderMap& request_headers) {
   absl::string_view method = request_headers.getMethodValue();
+  auto result = request_headers.get(Headers::get().EnvoyAllowEarlyData);
+  if (!result.empty() && result[0]->value() == "true") {
+    return true;
+  }
```

Comment:
I think it's not the perfect way to set a inner status in the request headers. It only bring additional performance penalty to our core code path which is what I don't want to see.